docs/intro/overview.md

```diff
@@ -46,54 +43,56 @@ But here is an (ES6) example of a view in the form of a React component.
-    render() {
-        return (
-            <button onClick={this.onReset.bind(this)}>
-                Seconds passed: {this.props.appState.timer}
-            </button>
-        )
-    }
-
-    onReset() {
-        this.props.appState.resetTimer()
-    }
-}
-
+const TimerView = observer(({ appState }) => (
+    <button onClick={() => appState.resetTimer()}>Seconds passed: {appState.timer}</button>
+))
 ReactDOM.render(<TimerView appState={appState} />, document.body)
 ```
 
-(For the implementation of `resetTimer` function see the next section)
+(For the implementation of `resetTimer` method see the next section)
 
 ## 3. Modify the State
 
 The third thing to do is to modify the state.
 That is what your app is all about after all.
-Unlike many other frameworks, MobX doesn't dictate how you do this.
-There are best practices, but the key thing to remember is:
-**_MobX helps you do things in a simple straightforward way_**.
 
-The following code will alter your data every second, and the UI will update automatically when needed.
-No explicit relations are defined in either the controller functions that _change_ the state or in the views that should _update_.
-Decorating your _state_ and _views_ with `observable` is enough for MobX to detect all relationships.
-Here are two examples of changing the state:
+Let's alter the timer every second, and see that the UI will update automatically when needed. Let's also implement that `resetTimer` method.
+
+Here's the new `AppState` model with a few methods added that modify state:
 
 ```javascript
-appState.resetTimer = action(function reset() {
-    appState.timer = 0
-})
-
-setInterval(
-    action(function tick() {
-        appState.timer += 1
-    }),
-    1000
-)
+import { makeObservable, observable, action } from "mobx"
+
+class AppState {
+    timer = 0
+
+    constructor() {
+        makeObservable(this, {
+            timer: observable,
+            resetTimer: action,
+            increaseTimer: action
+        })
+    }
+
+    increaseTimer() {
+        this.timer += 1
+    }
+
+    resetTimer() {
+        this.timer = 0
+    }
+}
+
+setInterval(() => {
+    appState.increaseTimer()
+}, 1000)
+
+const appState = new AppState()
 ```
 
-The `action` wrapper is only needed when using MobX in strict mode (by default off).
-It is recommended to use action though as it will help you to better structure applications and expresses the intention of a function to modify state.
-Also it automatically applies transactions for optimal performance.
+These methods, `increaseTimer` and `resetTimer` are just like you would write them without MobX. You can use them anywhere -- from React event handlers or in `setInterval`, for instance. The only thing you need to do is to mark them as `action` with `makeObservable`. By marking methods this way you make MobX automatically
+apply transactions for optimal performance.
+
+**_MobX helps you do things in a simple straightforward way_**.
```

Comment:
Might be nice to introduce `makeAutoObservable` at the end of this page, along the lines: we could have done this shorter by using `makeAutoObservable` and mobx will pick sane defaults

Reply:
I've now done this on the landing page but it makes sense here too. I think it's okay to have this duplication, so I'll add some verbiage here as well.